include/response_reader.rb

```diff
@@ -105,10 +105,10 @@ def url_and_args_from_location
     url << "?#{uri.query}" if uri.query.to_s.length > 0
     url = url.gsub(/\A\//, "")
 
-    args = {host: uri.host}
+    args = @http2.args
+           .except(:port)
```

Comment:
You're right, my bad - should have been consistent with the overrides.

Fixed in 87b8cde